fix(mqtt): fixed how mqtt library is imported

csarnataro opened the pull request:
Because of some issues with some bundlers (namely, Parcel 2) we have updated the way mqtt.connect is imported in Connection.ts

Minor updates to index.ts (eslint)

Added a changelog section in readme file

csarnataro committed:
fix(mqtt): fixed how mqtt library is imported

Because of some issues with some bundlers (namely, Parcel 2)
we have updated the way mqtt.connect is imported in Connection.ts

Minor updates to index.ts (eslint)

Added a changelog section in readme file

csarnataro committed:
fix(mqtt): fixed how mqtt library is imported

Minor fixes

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/arduino/arduino-iot-js?pullRequest=514) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/arduino/arduino-iot-js?pullRequest=514) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/arduino/arduino-iot-js?pullRequest=514) it.</sub>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/arduino/arduino-iot-js?pullRequest=514) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/arduino/arduino-iot-js?pullRequest=514) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/arduino/arduino-iot-js?pullRequest=514) it.</sub>

csarnataro committed:
fix(mqtt): fixed how mqtt library is imported

Regenerated package-lock.json

csarnataro committed:
fix(mqtt): fixed how mqtt library is imported

Updated TokenConnectionBuilder constructor to use the right mqtt.connect
function

fabioimpe commented on src/connection/Connection.ts:
```diff
@@ -106,7 +111,8 @@ export class Connection implements IConnection {
     });
 
-    if (valueToSend !== {}) messages.push({ topic, propertyName: current, value: valueToSend });
+    // the condition `if (valueToSend !== {}) ` has been removed bc it always evaluates to true
```
**thought**: maybe here is worth understanding if is better to remove the `if` or leaving the `if` fixing the condition

csarnataro replied:
Ok, it makes sense, I'm going to see if I can understand the original intention

csarnataro committed:
feat: added check on empy object

Added a check before sending MQTT empty objects.

csarnataro committed:
Bumped version

csarnataro commented on src/connection/Connection.ts:
```diff
@@ -106,7 +111,11 @@ export class Connection implements IConnection {
-    if (valueToSend !== {}) messages.push({ topic, propertyName: current, value: valueToSend });
+    // If the message is an object, and it's empty, it makes no sense to send it
+    // All other messages (e.g. non-empty objects or primitives) must be sent
+    if (!(typeof valueToSend === 'object' && Object.keys(valueToSend).length == 0)) {
```
@fabioimpe do you think it's ok now, with an explicit check on the object and its length?
I think now is more explicit that:
- if it's an empty object, skip sending the message
- if it's a non-empty object, or any other value (primitive) then send it.

I thing it's rather an edge-case, which probably happens if the `properties` array is empty (so `forEach` statement is never executed) or `value` at line 111 is an empty object.

fabioimpe replied:
Yes I think now is cleaner.
Maybe you can think about extracting the condition to be more readable, but this seems the right way 👍

csarnataro committed:
Minor refactoring, created a 'isNotAnEmptyObject' function

csarnataro committed:
Bumped version; added nvmrc file
